armi/reactor/geometry.py

```diff
@@ -113,6 +117,269 @@ def __str__(self):
+
+    @classmethod
+    def fromStr(cls, symmetryString: str) -> "SymmetryType":
+        symmetry = cls()
+        symmetry.shape = ShapeType.fromStr(symmetryString)
+        symmetry.boundary = BoundaryType.fromStr(symmetryString)
+        symmetry._checkIfThroughCenter(symmetryString)
+        return symmetry._returnIfValid()
+
+    @classmethod
+    def fromAny(cls, symmetry: Union[str, "SymmetryType"]) -> "SymmetryType":
+        """
+        Safely convert from string representation, no-op if already an enum instance.
+
+        This is useful as we transition to using enumerations more throughout the code.
+        There will remain situations where a geomType may be provided in string or enum
+        form, in which the consuming code would have to check the type before
+        proceeding. This function serves two useful purposes:
+         - Relieve client code from having to if/elif/else on ``isinstance()`` checks
+         - Provide a location to instrument these conversions for when we actually try
+           to deprecate the strings. E.g., produce a warning when this is called, or
+           eventually forbidding the conversion entirely.
+        """
+        if isinstance(symmetry, SymmetryType):
+            return symmetry._returnIfValid()
+        elif isinstance(symmetry, str):
+            return cls.fromStr(symmetry)
+        else:
+            raise TypeError("Expected str or SymmetryType; got {}".format(type(source)))
+
+    @classmethod
+    def fromSubTypes(
+        cls,
+        shapeType: Union[str, "ShapeType"],
+        boundaryType: Union[str, "BoundaryType"],
+        throughCenterAssembly: Optional[bool] = False,
+    ) -> "SymmetryType":
+
+        symmetry = cls()
+        symmetry.shape = ShapeType.fromAny(shapeType)
+        symmetry.boundary = BoundaryType.fromAny(boundaryType)
+        symmetry.isThroughCenterAssembly = throughCenterAssembly
+        return symmetry._returnIfValid()
+
+    def __str__(self):
+        """Combined string of shape and boundary symmetry type"""
+        strList = [str(self.shape)]
+        if self.boundary.hasSymmetry():
+            strList.append(str(self.boundary))
+        if self.isThroughCenterAssembly:
+            strList.append(THROUGH_CENTER_ASSEMBLY)
+        return _joinSpace(strList)
+
+    def _checkIfThroughCenter(self, symmetryString: str):
+        self.isThroughCenterAssembly = THROUGH_CENTER_ASSEMBLY in symmetryString
+
+    def _returnIfValid(self):
+        if self.checkValidSymmetry():
+            return self
+        else:
+            errorMsg = "{} is not a valid symmetry option. Valid symmetry options are:"
+            errorMsg += ", ".join([f"{sym}" for sym in VALID_SYMMETRY])
+            raise ValueError(errorMsg)
+
+    def symmetryFactor(self):
+        return self.shape.symmetryFactor()
+
+    def checkValidSymmetry(self):
+        return str(self) in VALID_SYMMETRY
+
+
+class ShapeType(enum.Enum):
```

Comment:
This could just be me, but I'm a little thrown off with shape type and geom type, since they kind feel like the same thing, just based on a) the words themselves, and b) that I'm used to thinking of GeomType as a shape. Maybe something like DomainType, since it's really referring to how much of the actual domain is represented? Thoughts?

Reply:
I didn't know what word to use exactly. Not sure what you'd call the property of full/third/quarter core other than "symmetry", which was already taken. Domain seems like a good fit.